editor/plugins/theme_editor_plugin.cpp

```diff
@@ -3664,6 +3665,59 @@ void ThemeEditor::_preview_control_picked(String p_class_name) {
 	theme_type_editor->select_type(p_class_name);
 }
 
+bool ThemeEditor::can_drop_data_fw(const Point2 &p_point, const Variant &p_data, Control *p_from) {
+	Dictionary d = p_data;
+	if (!d.has("type")) {
+		return false;
+	}
+
+	if (String(d["type"]) == "files") {
+		Vector<String> files = d["files"];
+
+		if (files.size() != 1) {
+			return false;
+		}
+
+		String ftype = EditorFileSystem::get_singleton()->get_file_type(files[0]);
+		if (ftype != "PackedScene") {
+			return false;
+		}
+		return true;
+	}
+	return false;
+}
+
+void ThemeEditor::drop_data_fw(const Point2 &p_point, const Variant &p_data, Control *p_from) {
+	if (!can_drop_data_fw(p_point, p_data, p_from)) {
+		return;
+	}
+
+	Dictionary d = p_data;
+	if (String(d["type"]) == "files") {
+		Vector<String> files = d["files"];
+		const String &path = files[0];
+
+		SceneThemeEditorPreview *preview_tab = memnew(SceneThemeEditorPreview);
+		if (!preview_tab->set_preview_scene(path)) {
+			return;
```

Comment:
This will leak `preview_tab`. You should `memfree()` it.
(I noticed that you copied it; the other code is wrong too)

Reply:
Actually it should be `memdelete()`, we use it more often.